backend/src/routers/user.ts

```diff
@@ -101,32 +100,27 @@ userRouter.post('/logout/', (req, res) =>
     res.cookie(tokenCookieName, '-', tokenCookieOptions).status(204).end()
 );
 
-userRouter.get(
-    '/',
-    expressAsyncHandler(async (req, res) => {
-        const payload = await readToken(req);
-        if (payload === null) {
-            res.status(401).send('No valid token on the request found');
-        } else {
-            const response = {
-                name: payload.userName,
-                id: payload.userID,
-            };
-            res.json(response);
-        }
-    })
-);
+userRouter.get('/', (req, res) => {
+    if (req.token === null) {
```

Comment:
Wasn't the point of the middleware so that we can skip this repetetive token check in every function that requires it? Couldn't every api point with a if (req.token === null) { return 401 } be replaced by the checkToken middleware?

Reply:
The problem with this router is that some of the endpoints must be available for anonymous users and therefore we can not use the checkToken middleware. Alternatively we could split this router into two different routers and place one before and the other one after the middleware. However, that may get a bit messy with the routing paths.